hack/vendor-console.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -o errexit 
+set -o nounset 
+set -o pipefail
+
+STARTTIME=$(date +%s)
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${OS_ROOT}/hack/common.sh"
+
+CONSOLE_REPO_PATH=${CONSOLE_REPO_PATH:-$OS_ROOT/../origin-web-console}
+if [ ! -d "$CONSOLE_REPO_PATH" ]; then
+  echo "The console repo at path $CONSOLE_REPO_PATH does not exist."
+  echo "Make sure you have cloned the web console repo locally:  git@github.com:openshift/origin-web-console.git"
+  echo "Or, you can specify a different path with the CONSOLE_REPO_PATH environment variable."
+  exit 1
+fi
+
+echo "Making sure go-bindata binary has been built..."
+pushd "${OS_ROOT}/Godeps/_workspace" > /dev/null
+  godep_path=$(pwd)
+  pushd src/github.com/jteeuwen/go-bindata > /dev/null
+    GOPATH=$godep_path go install ./...
+  popd > /dev/null
+popd > /dev/null
+
+if [[ -z "${GIT_REF:+x}" ]]; then 
+  echo "No git ref specified, using current state of the repo including any unstaged changes...";
+else 
+  echo "Using git ref ${GIT_REF}, all untracked and uncommitted changes will be stashed..."
+  # TODO improve this to return the user back to the state they were in
+  pushd "${CONSOLE_REPO_PATH}" > /dev/null
+    git stash --include-untracked
+    git checkout $GIT_REF
```

Comment:
If you are expecting the splitting of args on spaces here and not quoting to allow for that. leave a comment as it's the unusual case